lib/ts/fetch.ts

```diff
@@ -134,7 +134,11 @@ export class FrontToken {
 
     static async setItem(frontToken: string) {
         // We update the refresh attempt info here as well, since this means that we've updated the session in some way
-        // maybe through a custom endpoint...
+        // This could be both by a refresh call or if the access token was updated in a custom endpoint
+        // By saving every time the access token has been updated, we cause an early retry if
+        // another request has failed with a 401 with the previous access token and the token still exists.
```

Comment:
- Why do we need to cause an early retry? If refreshing is taking place, and during that, the access token was updated, it could still mean that the access token is expired cause updating the access token payload doesn't change it's expiry.
- If we must do this, then we should probably change the `saveRefreshAttempt` function to something like `saveAccessTokenUpdateTime` -> this would store the last time the access token was updated or refreshed (it;'s a better name given the context).

Reply:
- if request A gets a 401 and starts to acquire a lock, while request B updates the access token, it means that the access token is not expired since B could update it. If request B is executed on the server before A and the token expires between the two, then we make an extra retry call, but this should be rare enough.
- Without this, we'd miss user APIs creating (and, in rarer cases, refreshing) sessions if using cookies since we'd only see the front-token update.

The basic idea is that we knew we had an expired token, but that information is outdated since the token we'd be using for the retry isn't the same.